### DISCUSSION TITLE
handle variable/missing columns in parquet with a default "fillna"?

### DISCUSSION THREAD

**Discussion by cottrell** [duckdb/duckdb#4482] [Q&A]
Consider a bunch of parquet where some have the column 'd' and some do not.

It seems you *can* query these tables if you explicitly avoid using the column d.

Is there some nice way of specifying a schema where 'd' gets filled with a default if it doesn't exist?

Something like this
```python
import numpy as np
import pandas as pd
import duckdb

dfa = pd.DataFrame(np.random.randn(5, 3), columns=['a', 'b', 'c'])
dfb = pd.DataFrame(np.random.randn(5, 4), columns=['a', 'b', 'c', 'd'])
dfa.to_parquet('dfa.parquet')
dfb.to_parquet('dfb.parquet')


con = duckdb.connect()
print(con.execute("select a, b, c from parquet_scan('df*.parquet')").df().shape)
try:
    print(con.execute("select * from parquet_scan('df*.parquet')").df().shape)
    print('this should have failed')
except Exception as e:
    print('yes this fails')
```

**Comment by Alex-Monahan** [comment]
Hmmm, could the new Union by name approach get used under the hood here moving forward?

**Reply by cottrell** [reply]
Would be interesting in this ... I'm guessing a bit of work under the hood inspecting the meta-data (schema) to for a pre-query is needed. If anyone knows more would be interested in being pointed to where this could happen in the code or if it makes sense. I also saw some comment somewhere about UNION ALL breaking parallelism but maybe that is no longer true.

**Reply by Alex-Monahan** [reply]
Union all is now parallelized!
